src/string.cr

```diff
@@ -1742,6 +1742,18 @@ class String
     end
   end
 
+  # Returns a string where all chars in the given hash are replaced
+  # by the corresponding *tuple* values.
+  #
+  # ```
+  # "hello".gsub({'e' => 'a', 'l' => 'd'}) # => "haddo"
+  # ```
+  def gsub(tuple : NamedTuple)
+    gsub do |char|
+      tuple[char.to_s]? || char
```

Comment:
Should we have`Tuple#fetch` to allow proper non-fallbacking when the actual value is falsy?
Not particularly needed in this case, but It's been a week with `hash[key] || default` bugs on my side :-)

Maybe adding `Indexable#fetch`?
`Hash#fetch` currently uses `#find_entry`, so I don't expect to promote that method to the module.